netplugin/plugin/netplugin.go

```diff
@@ -151,11 +151,15 @@ func (p *NetPlugin) DeleteEndpoint(id string) error {
 
 // CreateRemoteEndpoint creates an endpoint for a given ID.
 func (p *NetPlugin) CreateRemoteEndpoint(id string) error {
+	p.Lock()
+	defer p.Unlock()
 	return p.NetworkDriver.CreateRemoteEndpoint(id)
 }
 
 // DeleteRemoteEndpoint destroys an endpoint for an ID.
 func (p *NetPlugin) DeleteRemoteEndpoint(id string) error {
+	p.Lock()
```

Comment:
Isnt the lock acquired here also ?  https://github.com/contiv/netplugin/blob/b6719ea52b1ee96cd13016a868ada555bfe545eb/netplugin/agent/state_event.go#L279